src/v/config/node_config.cc

```diff
@@ -167,6 +167,12 @@ node_config::node_config() noexcept
       "scrubber",
       {.visibility = visibility::user},
       std::nullopt)
+  , datalake_staging_directory(
+      *this,
+      "datalake_staging_directory",
+      "Directory for datalake temporary storage.",
```

Comment:
nit: mention that if it's not set, the default is to place data in the data directory?